lib/pf/Switch/Cisco/WLC_http.pm

```diff
@@ -163,8 +163,9 @@ sub returnRadiusAccessAccept {
     }
 
 
-    # if Roles aren't configured, return VLAN information
-    if (isenabled($this->{_VlanMap}) ) {
+    # if vlan assignement is activated, then we use it for 802.1x connections
+    # we don't return a VLAN for MAC auth since that's why the other WLC modules are there
+    if (isenabled($this->{_VlanMap}) && !($connection_type eq $WIRELESS_MAC_AUTH)) {
```

Comment:
ok for that but in the case before:
if (defined($role) && isenabled($this->{_RoleMap})) {
don´t we have to do that ? :
if (defined($role) && isenabled($this->{_RoleMap} && ($connection_type eq $WIRELESS_MAC_AUTH) ) {